pkg/planner/core/logical_plan_builder.go

```diff
@@ -250,6 +250,7 @@ func (b *PlanBuilder) buildAggregation(ctx context.Context, p base.LogicalPlan,
 	b.optFlag |= flagPredicatePushDown
 	b.optFlag |= flagEliminateAgg
 	b.optFlag |= flagEliminateProjection
+	b.optFlag |= flagConvertOuterToInnerJoin
```

Comment:
Do we need to add this flag in buildAgg ? This is only a rule for join node, so maybe we can just flag this rule in buildJoin ?

Reply:
Outer to inner has its own push down through nodes and does not rely on predicate push down. if we limit it to joins then we may miss out something like 
with cte as (select count(a2) as cnt,b2 from t1 left outer join t2 on a1=a2 group by b2) select * from cte where b2 > 1

Reply:
Oh, I understand. We should judge the predicate which is null-rejected firstly furthermore we can judge “outer join to inner join”.
But if we don't have join clause we don't need to judge the predicate which is null-rejected.
So it can cover all situations when we just mark the flag at the "buildjoin" . I don’t know if I understand it correctly?

Reply:
I see your point. At the build level, for aggregations there is no predicate to pass down for null check. But, why flagPredicatePushDown is used here? I basically, followed the logic of flagPredicatePushDown in setting the flagConvertOuterToInnerJoin.